lms/views/admin.py

```diff
@@ -0,0 +1,91 @@
+
+from lms.security import Permissions
+
+
+@view_defaults(request_method="GET")
+class AdminViews:
+    def __init__(self, request):
+        self.request = request
+        self.application_instance = request.find_service(name="application_instance")
+
+    @view_config(
+        route_name="admin.index",
+        renderer="lms:templates/admin/index.html.jinja2",
+        permission=Permissions.ADMIN,
+    )  # pylint: disable=no-self-use
+    def index(self):
+        return {}
+
+    @view_config(
+        route_name="admin.installations",
+        renderer="lms:templates/admin/installations.html.jinja2",
+        permission=Permissions.ADMIN,
+    )  # pylint: disable=no-self-use
+    def installations(self):
+        return {}
+
+    @view_config(
+        route_name="admin.installations",
+        permission=Permissions.ADMIN,
+        request_method="POST",
+    )
+    def find_installation(self):
+        if "query" not in self.request.params:
+            self.request.session.flash("Missing mandatory 'query'", "errors")
+            return HTTPFound(location=self.request.route_url("admin.installations"))
+
+        installation = self.application_instance.find(self.request.params["query"])
+        if installation:
+            return HTTPFound(
+                location=self.request.route_url(
+                    "admin.installation", id=installation.id
+                ),
+            )
+        self.request.session.flash(
+            f'No installation found for {self.request.params["query"]}', "errors"
+        )
+        return HTTPFound(location=self.request.route_url("admin.installations"))
+
+    @view_config(
+        route_name="admin.installation",
+        renderer="lms:templates/admin/installation.html.jinja2",
+        permission=Permissions.ADMIN,
+    )
+    def show_installation(self):
+
+        installation = self.application_instance.get(self.request.matchdict["id"])
+        return {"installation": installation}
+
+    @view_config(
+        route_name="admin.installation",
+        permission=Permissions.ADMIN,
+        request_method="POST",
+    )
+    def update_installation(self):
+        installation = self.application_instance.get(self.request.matchdict["id"])
+
+        sections_enabled = (
+            "sections_enabled" in self.request.params
+            and self.request.params["sections_enabled"] == "on"
+        )
+        groups_enabled = (
+            "groups_enabled" in self.request.params
+            and self.request.params["groups_enabled"] == "on"
+        )
+
+        self.application_instance.update_settings(
+            installation,
+            canvas_sections_enabled=sections_enabled,
+            canvas_groups_enabled=groups_enabled,
+        )
+        self.request.session.flash(
+            f"Updated installation {installation.id}", "messages"
+        )
+
+        return HTTPFound(location=self.request.route_url("admin.installations"))
```

Comment:
It'd be nice if this redirected to `route_url(admin.installation`, id=installation.id)` instead so I could see the checkboxes reloaded with their new values